### PR TITLE
chore(repo): Test task cache, instance cleanup

### PR DESCRIPTION
**Pull request by BRKalow**
## Description

Fixes caching for integration test tasks, all task dependencies incorrectly contained `^`, which means the task should depend on its package's dependencies running the tasks specified with `^`. This was effectively a no-op when used for a root task with package-specific tasks in its dependencies. This should fix integration test caching 🤞

Also fixes an issue with the integration test cleanup script where not all instances were being cleaned up.

## Checklist

- [ ] `npm test` runs as expected.
- [ ] `npm run build` runs as expected.
- [ ] (If applicable) [JSDoc comments](https://jsdoc.app/about-getting-started.html) have been added or updated for any package exports
- [ ] (If applicable) [Documentation](https://github.com/clerk/clerk-docs) has been updated

## Type of change

- [ ] 🐛 Bug fix
- [ ] 🌟 New feature
- [ ] 🔨 Breaking change
- [ ] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

### CODE REVIEW

**Review by panteliselef — APPROVED** [review]
What's the reason for dropping `apiUrl`. Unused property ?

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -139,6 +139,7 @@ jobs:
     timeout-minutes: ${{ vars.TIMEOUT_MINUTES_LONG && fromJSON(vars.TIMEOUT_MINUTES_LONG) || 15 }}
 
     strategy:
+      fail-fast: false
```
Comment (BRKalow) [inline .github/workflows/ci.yml:142]: Given the flakiness of our integration tests, it feels valuable to continue running tests even if others fail. Many times tests are minutes into running when cancelled, which I don't think has the cost savings we expect in practice.

File: integration/cleanup/cleanup.setup.ts [hunk 2]
```diff
@@ -5,20 +5,19 @@ import { test as setup } from '@playwright/test';
 import { appConfigs } from '../presets/';
 
 setup('cleanup instances ', async () => {
-  const entries = Object.values(appConfigs.envs)
```
Comment (BRKalow) [inline integration/cleanup/cleanup.setup.ts:8]: `appConfig.envs` didn't contain all the keys in use, so not all instances were getting cleaned up as far as I could tell.

File: turbo.json [hunk 3]
```diff
@@ -136,84 +136,79 @@
       "outputs": []
     },
     "//#test:integration:ap-flows": {
-      "dependsOn": ["^@clerk/clerk-js#build", "^@clerk/backend#build", "^@clerk/nextjs#build"],
+      "dependsOn": ["@clerk/clerk-js#build", "@clerk/backend#build", "@clerk/nextjs#build"],
```
Comment (BRKalow) [inline turbo.json:139]: All of the adjustments here remove the `^` for each dependency. The `^` says to wait on the package's dependencies, which for a root task like this effectively seems like a no-op.

I've validated that this properly caches the integration test run only if the dependent build steps remain unchanged, which is the desired behavior.

**Review by panteliselef — APPROVED** [review]

**Pull request merged**